src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java

```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
 	public void verifyLightboxClosed() {
 		Assertion.assertFalse(checkIfElementOnPage(currentImage));
 	}
+	
+	public String currentImageSrcPath () {
+		return currentImage.getAttribute("src");
+	}
+	
+	public void verifySwiping (PerformTouchAction touchAction, String direction, int attempts) {
+		String currentImageSrc = getCurrentImagePath();
+		String nextImageSrc;
+		boolean imageChanged = false;
+		int startX = 70;
+		int endX = 20;
+		int duration = 300;
+		int waitAfter = 5000;
+		if (direction == "right") {
+			int temp = startX;
+			startX = endX;
+			endX = temp;
+		} else {
+			direction = "left";
+		}
+		for (int i = 0; i < attempts; ++i) {
+			touchAction.swipeFromPointToPoint(startX, 50, endX, 50, duration, waitAfter);
+			nextImageSrc = getCurrentImagePath();
+			if (!nextImageSrc.contains(currentImageSrc)) {
+				imageChanged = true;
+				break;
+			}
+		}
+		if (imageChanged) {
+			PageObjectLogging.log("verifySwiping", "Swiping to " + direction + " works", true);
+		} else {
+			PageObjectLogging.log("verifySwiping", "Swiping to " + direction + " doesn't work", false);
+		}
+	}
+	
+	private boolean verifyChangesInUI (boolean lastDisplayVisibility) {
+		String lightboxHeaderDisplayValue = lightboxHeader.getCssValue("display");
+		String lightboxFooterDisplayValue = lightboxFooter.getCssValue("display");
+		boolean lightboxHeaderDisplay;
+		boolean lightboxFooterDisplay;
+		if (lightboxHeaderDisplayValue.contains("none")) {
+			lightboxHeaderDisplay = false;
+		} else {
+			lightboxHeaderDisplay = true;
+		}
+		if (lightboxFooterDisplayValue.contains("none")) {
+			lightboxFooterDisplay = false;
+		} else {
+			lightboxFooterDisplay = true;
+		}
+		if (lightboxHeaderDisplay == lastDisplayVisibility) {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox header visibility didn't changed", false);
+		} else {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox header visibility changed", true);
+		}
+		if (lightboxFooterDisplay == lastDisplayVisibility) {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox footer visibility didn't changed", false);
+		} else {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox footer visibility changed", true);
+		}
+		return !lastDisplayVisibility;
+	}
+	
+	public void verifyVisibilityUI (PerformTouchAction touchAction) {
+		String LightboxHeaderDisplayValue = lightboxHeader.getCssValue("display");
+		String LightboxFooterDisplayValue = lightboxFooter.getCssValue("display");
+		int duration = 500;
+		int waitAfter = 5000;
+		boolean lastDisplayVisibility = true;
+		if (LightboxHeaderDisplayValue.contains(LightboxFooterDisplayValue)) {
+			if (LightboxHeaderDisplayValue == "none") {
```

Comment:
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException